versions/3.0.4.md

```diff
@@ -1609,19 +1609,51 @@ An `encoding` attribute is introduced to give you control over the serialization
+
+Property Type | Property Format | Default `contentType`
+------------- | --------------- | ---------------------
+`string` | `binary` | `application/octet-stream`
+`string`, `number`, `integer`, or `boolean` | _n/a_ | `text/plain`
+`object` | _n/a_ | `application/json`
+`array` | _n/a_ | according to the `type` and `format` of the `items` schema
+
+###### Encoding `multipart` Media Types
+
+The `form-data` disposition and its `name` parameter that is used to correlate parts with encodings are mandatory for `multipart/form-data` ([RFC7578 §4.2](https://www.rfc-editor.org/rfc/rfc7578.html#section-4.2)).
+Array properties are handled by applying the same `name` to multiple parts, as is recommended by [RFC7578 §4.2](https://www.rfc-editor.org/rfc/rfc7578.html#section-4.3) for supplying multiple values per form field.
+See [RFC7578 §5](https://www.rfc-editor.org/rfc/rfc7578.html#section-5) for guidance regarding non-ASCII part names.
+
+Various other `multipart` types, most notable `multipart/mixed` ([RFC2046 §5.1.3](https://www.rfc-editor.org/rfc/rfc2046.html#section-5.1.3)) neither require nor forbid specific `Content-Disposition` values, which means care must be taken to ensure that any values used are supported by all relevant software.
+It is not currently possible to correlate schema properties with unnamed, ordered parts in media types such as `multipart/mixed`, but implementations MAY choose to support such types when `Content-Disposition: form-data` is used with a `name` parameter.
+
+Note that there are significant restrictions on what headers can be used with `multipart` media types in general ([RFC2046 §5.1](https://www.rfc-editor.org/rfc/rfc2046.html#section-5.1)) and `multi-part/form-data` in particular ([RFC7578 §4.8](https://www.rfc-editor.org/rfc/rfc7578.html#section-4.8)).
+
+Note also that `Content-Transfer-Encoding` is deprecated for `multipart/form-data` ([RFC7578 §4.7](https://www.rfc-editor.org/rfc/rfc7578.html#section-4.7)) where binary data is supported, as it is in HTTP.
+Using `format: byte` for a multipart field is equivalent to setting `Content-Transfer-Encoding: base64`.
+If `format: byte` is used along with setting a different `Content-Transfer-Encoding` value with the `headers` field, the result is undefined.
+
+###### Encoding the `application/x-www-form-urlencoded` Media Type
+
+The role of `contentType` with `application/x-www-form-urlencoded` request bodies was not described in detail in version 3.0.3 and earlier of this specification.
+To match the intent of these fields and be compatible with version 3.1 of this specification, it is RECOMMENDED that whenever any of `style`, `explode`, or `allowReserved` are present with an explicit values, that the value of `contentType`, whether it is explicitly defined or has the default value, be ignored.
+Likewise, if all three of those fields are absent, it is RECOMMENDED that they be ignored, and that encoding be based on `contentType` alone.
```

Comment:
I'm not sure what it means to ignore these fields when they are not present. Ignore any default value, maybe?

Reply:
Yeah, that's what it means, I'll improve the wording.  Thanks!  This is a weird thing to even have to explain, and has to do with how the Encoding Object was kind-of spliced together from ideas that weren't designed together.

Reply:
OK I ended up reworking this section a bit to make all of the default / explicit interactions more clear.

@notEthan please let me know what you think of this version!